// @clutch/helpers/dist/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  addKey: true,
  asyncQueue: true,
  calculateElementAttribute: true,
  callAll: true,
  createValidationFunction: true,
  emptyStringWhenLessThanZero: true,
  extractKey: true,
  extractValueFromEvent: true,
  formatMileage: true,
  formatPhoneNumber: true,
  formatPrice: true,
  includes: true,
  isTruthy: true,
  isValidNumber: true,
  numberMask: true,
  persistEvent: true,
  preventDefault: true,
  reconstructEvent: true,
  removeNonNumbers: true,
  snakeCaseToReadableString: true,
  throwError: true,
  truthyOrZero: true,
  zeroIfEmptyString: true
};
Object.defineProperty(exports, "addKey", {
  enumerable: true,
  get: function get() {
    return _addKey["default"];
  }
});
Object.defineProperty(exports, "asyncQueue", {
  enumerable: true,
  get: function get() {
    return _asyncQueue["default"];
  }
});
Object.defineProperty(exports, "calculateElementAttribute", {
  enumerable: true,
  get: function get() {
    return _calculateElementAttribute["default"];
  }
});
Object.defineProperty(exports, "callAll", {
  enumerable: true,
  get: function get() {
    return _callAll["default"];
  }
});
Object.defineProperty(exports, "createValidationFunction", {
  enumerable: true,
  get: function get() {
    return _createValidationFunction["default"];
  }
});
Object.defineProperty(exports, "emptyStringWhenLessThanZero", {
  enumerable: true,
  get: function get() {
    return _emptyStringWhenLessThanZero["default"];
  }
});
Object.defineProperty(exports, "extractKey", {
  enumerable: true,
  get: function get() {
    return _extractKey["default"];
  }
});
Object.defineProperty(exports, "extractValueFromEvent", {
  enumerable: true,
  get: function get() {
    return _extractValueFromEvent["default"];
  }
});
Object.defineProperty(exports, "formatMileage", {
  enumerable: true,
  get: function get() {
    return _formatMileage["default"];
  }
});
Object.defineProperty(exports, "formatPhoneNumber", {
  enumerable: true,
  get: function get() {
    return _formatPhoneNumber["default"];
  }
});
Object.defineProperty(exports, "formatPrice", {
  enumerable: true,
  get: function get() {
    return _formatPrice["default"];
  }
});
Object.defineProperty(exports, "includes", {
  enumerable: true,
  get: function get() {
    return _includes["default"];
  }
});
Object.defineProperty(exports, "isTruthy", {
  enumerable: true,
  get: function get() {
    return _isTruthy["default"];
  }
});
Object.defineProperty(exports, "isValidNumber", {
  enumerable: true,
  get: function get() {
    return _isValidNumber["default"];
  }
});
Object.defineProperty(exports, "numberMask", {
  enumerable: true,
  get: function get() {
    return _numberMask["default"];
  }
});
Object.defineProperty(exports, "persistEvent", {
  enumerable: true,
  get: function get() {
    return _persistEvent["default"];
  }
});
Object.defineProperty(exports, "preventDefault", {
  enumerable: true,
  get: function get() {
    return _preventDefault["default"];
  }
});
Object.defineProperty(exports, "reconstructEvent", {
  enumerable: true,
  get: function get() {
    return _reconstructEvent["default"];
  }
});
Object.defineProperty(exports, "removeNonNumbers", {
  enumerable: true,
  get: function get() {
    return _removeNonNumbers["default"];
  }
});
Object.defineProperty(exports, "snakeCaseToReadableString", {
  enumerable: true,
  get: function get() {
    return _snakeCaseToReadableString["default"];
  }
});
Object.defineProperty(exports, "throwError", {
  enumerable: true,
  get: function get() {
    return _throwError["default"];
  }
});
Object.defineProperty(exports, "truthyOrZero", {
  enumerable: true,
  get: function get() {
    return _truthyOrZero["default"];
  }
});
Object.defineProperty(exports, "zeroIfEmptyString", {
  enumerable: true,
  get: function get() {
    return _zeroIfEmptyString["default"];
  }
});

var _addKey = _interopRequireDefault(require("./addKey"));

var _asyncQueue = _interopRequireDefault(require("./async-queue"));

var _calculateElementAttribute = _interopRequireDefault(require("./calculateElementAttribute"));

var _callAll = _interopRequireDefault(require("./callAll"));

var _createValidationFunction = _interopRequireDefault(require("./createValidationFunction"));

var _emptyStringWhenLessThanZero = _interopRequireDefault(require("./emptyStringWhenLessThanZero"));

var _extractKey = _interopRequireDefault(require("./extractKey"));

var _extractValueFromEvent = _interopRequireDefault(require("./extractValueFromEvent"));

var _formatMileage = _interopRequireDefault(require("./formatMileage"));

var _formatPhoneNumber = _interopRequireDefault(require("./formatPhoneNumber"));

var _formatPrice = _interopRequireDefault(require("./formatPrice"));

var _includes = _interopRequireDefault(require("./includes"));

var _isTruthy = _interopRequireDefault(require("./isTruthy"));

var _isValidNumber = _interopRequireDefault(require("./isValidNumber"));

var _numberMask = _interopRequireDefault(require("./numberMask"));

var _persistEvent = _interopRequireDefault(require("./persistEvent"));

var _preventDefault = _interopRequireDefault(require("./preventDefault"));

var _reconstructEvent = _interopRequireDefault(require("./reconstructEvent"));

var _removeNonNumbers = _interopRequireDefault(require("./removeNonNumbers"));

var _snakeCaseToReadableString = _interopRequireDefault(require("./snakeCaseToReadableString"));

var _throwError = _interopRequireDefault(require("./throwError"));

var _truthyOrZero = _interopRequireDefault(require("./truthyOrZero"));

var _zeroIfEmptyString = _interopRequireDefault(require("./zeroIfEmptyString"));

var _validation = require("./validation");

Object.keys(_validation).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _validation[key];
    }
  });
});
